src/eval/helpers.rs

```diff
@@ -72,15 +72,38 @@ pub fn call_js_function<'s>(
     args: &[v8::Local<v8::Value>],
 ) -> Result<v8::Local<'s, v8::Value>, NixError> {
     let try_scope = &mut v8::TryCatch::new(scope);
-    let recv = v8::undefined(try_scope).into();
-    let Some(strict_nix_value) = js_function.call(try_scope, recv, args) else {
-        // TODO: Again, the stack trace needs to be source-mapped.
-        if let Some(error) = try_scope.exception() {
-            let error = js_error_to_rust(try_scope, nixrt, error);
-            return Err(error);
-        } else {
-            return Err("Unknown evaluation error.".into());
-        }
+    let this = v8::undefined(try_scope).into();
+    let Some(strict_nix_value) = js_function.call(try_scope, this, args) else {
+        let exception = try_scope.exception();
+        return Err(map_js_exception_value_to_rust(try_scope, nixrt, exception));
     };
     Ok(strict_nix_value)
 }
+
+pub fn call_applied_js_function<'s>(
```

Comment:
Not sure what "applied function" means. Does it mean "instance method"? If so, could you please rename to:
```typescript
pub fn call_js_instance_mehod<'s>(
```

If  not, could you add a comment explaining what an "applied" function is?

Reply:
Yeah I had a bit of a mind blank with this one. You can `.apply` on functions in js, but yeah this is just an instance method.